src/Bridge/Doctrine/Common/Filter/AbstractSearchFilter.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Common\Filter;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\DBAL\Types\Type;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * Abstract class for filtering the collection by given properties.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+abstract class AbstractSearchFilter extends AbstractContextAwareFilter
+{
+    /**
+     * @var string Exact matching
+     */
+    const STRATEGY_EXACT = 'exact';
+
+    /**
+     * @var string The value must be contained in the field
+     */
+    const STRATEGY_PARTIAL = 'partial';
+
+    /**
+     * @var string Finds fields that are starting with the value
+     */
+    const STRATEGY_START = 'start';
+
+    /**
+     * @var string Finds fields that are ending with the value
+     */
+    const STRATEGY_END = 'end';
+
+    /**
+     * @var string Finds fields that are starting with the word
+     */
+    const STRATEGY_WORD_START = 'word_start';
+
+    protected $iriConverter;
+    protected $propertyAccessor;
+
+    /**
+     * @param RequestStack|null $requestStack No prefix to prevent autowiring of this deprecated property
```

Comment:
Injecting `$requestStack` is deprecated now, so if we create new classes, we shouldn't add it to the condtructor signature.